iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/FileUploadTests.java

```diff
@@ -87,46 +86,7 @@
     protected static final String testProxyUser = "proxyUsername";
     protected static final char[] testProxyPass = "1234".toCharArray();
 
-    static Set<FileUploadNotification> activeFileUploadNotifications = new ConcurrentSkipListSet<>(new Comparator<FileUploadNotification>()
-    {
-        @Override
-        public int compare(FileUploadNotification o1, FileUploadNotification o2) {
-            if (!o1.getDeviceId().equals(o2.getDeviceId()))
-            {
-                return -1;
-            }
-
-            if (!o1.getBlobName().equals(o2.getBlobName()))
-            {
-                return -1;
-            }
-
-            if (!o1.getBlobSizeInBytes().equals(o2.getBlobSizeInBytes()))
-            {
-                return -1;
-            }
-
-            if (!o1.getBlobUri().equals(o2.getBlobUri()))
-            {
-                return -1;
-            }
-
-            if (!o1.getEnqueuedTimeUtcDate().equals(o2.getEnqueuedTimeUtcDate()))
-            {
-                return -1;
-            }
-
-            if (!o1.getLastUpdatedTimeDate().equals(o2.getLastUpdatedTimeDate()))
-            {
-                return -1;
-            }
-
-            return 0;
-        }
-    });
-    static Thread fileUploadNotificationListenerThread;
-    static AtomicBoolean hasFileUploadNotificationReceiverThreadFailed = new AtomicBoolean(false);
-    static Exception fileUploadNotificationReceiverThreadException = null;
```

Comment:
Instead of running a thread in parallel to all the test methods to collect file upload notifications, now each test method just collects file upload notifications in their own thread. They still pool all the received notifications in the queue above so that if test method A receives a notification that test method B needs, test method B can still get it.

Reply:
This separate thread was causing issues when running these tests on Android, so it had to go away anyways